### PR TITLE
Firestore config loader

### PR DESCRIPTION
**Pull request by vmax**

### CODE REVIEW

File: android/BUILD [hunk 1]
```diff
@@ -44,17 +28,23 @@ android_library(
...
-    ],
+    ]),
     custom_package = "com.google.startupos.android",
     manifest = "src/main/java/com/google/startupos/android/AndroidManifest.xml",
     resource_files = glob(["src/main/java/com/google/startupos/android/res/**"]),
     visibility = ["//visibility:public"],
-    deps = [":firestore"],
+    assets = ["src/main/assets/google-services.json"],
+    assets_dir = "src/main/assets/",
+    deps = [":firestore", ":firestore_config_manager"],
```
Comment (oferb) [inline android/BUILD]: Please separate deps to separate lines (like srcs).

File: android/src/main/java/com/google/startupos/android/MainApplication.java [hunk 2]
```diff
@@ -17,15 +17,29 @@
...
 
 public class MainApplication extends Application {
+
+  public static final String TAG = MainApplication.class.getSimpleName();
+
   @Override
   public void onCreate() {
     super.onCreate();
-    FirebaseApp.initializeApp(this, FirestoreConfig.getFirebaseConfig());
+    try {
```
Comment (oferb) [inline android/src/main/java/com/google/startupos/android/MainApplication.java]: Why did we drop FirebaseApp.initializeApp()?

File: android/src/main/java/com/google/startupos/android/MainApplication.java [hunk 3]
```diff
@@ -17,15 +17,29 @@
...
+      FirestoreConfigLoader.load(this);
+    } catch (IOException e) {
+      Log.e(TAG, "Unable to initialize Firebase: no google-services.json provided");
```
Comment (oferb) [inline android/src/main/java/com/google/startupos/android/MainApplication.java]: Rephrase:
Firebase init failed: google-services.json not found

File: android/src/main/java/com/google/startupos/android/MainApplication.java [hunk 4]
```diff
@@ -17,15 +17,29 @@
...
+    } catch (JSONException e) {
+      Log.e(TAG, "Unable to initialize Firebase: google-services.json has incorrect structure");
```
Comment (oferb) [inline android/src/main/java/com/google/startupos/android/MainApplication.java]: Firebase init failed: google-services.json has incorrect structure

File: android/src/main/assets/google-services.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
+{
+  "project_info": {
+    "project_number": "606957953509",
+    "firebase_url": "https://farmers-tmp.firebaseio.com",
```
Comment (oferb) [inline android/src/main/assets/google-services.json]: farmers-tmp? :)
How about you open a new Firestore project for this demo app (startupos-android-example)?

Or maybe we shouldn't include a google-services.json. WDYT?

File: android/src/main/java/com/google/startupos/android/MainApplication.java (hunk 2 above)
Comment (vmax) [reply android/src/main/java/com/google/startupos/android/MainApplication.java]: @oferb we now delegate initialization to config loader

File: android/src/main/assets/google-services.json (hunk 5 above)
Comment (vmax) [reply android/src/main/assets/google-services.json]: @oferb 
>Or maybe we shouldn't include a google-services.json. WDYT?
it got into PR by mistake 🙂

File: android/src/main/java/com/google/startupos/android/MainApplication.java (hunk 2 above)
Comment (oferb) [reply android/src/main/java/com/google/startupos/android/MainApplication.java]: If this also initializes and not only loads the config, let's call it FirestoreInitializer with init()

File: android/src/main/assets/google-services.json (hunk 5 above)
Comment (oferb) [reply android/src/main/assets/google-services.json]: How about we add it to .gitignore then?

**Pull request merged**